src/NiceWrapper.cpp

```diff
@@ -227,6 +237,21 @@ void NiceWrapper::Stop() {
   }
 }
 
+void NiceWrapper::ParseRemoteSDP(std::string remote_sdp) {
+  string crfree_remote_sdp = remote_sdp;
+
+  // TODO: Improve this. This is needed because otherwise libnice will wrongly take the '\r' as part of ice-ufrag/password.

```

Comment:
I hear you. I will dig into this a bit more at some point. It appears that libnice would rather deal with just "\n" everywhere, but maybe the "\r\n" issue only reveals itself for the ice-ufrag/password .... If that's the case, we can enforce that properly and convert "\r\n" to "\n" at the NiceWrapper level....